Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -432,17 +487,27 @@ class Gx2Fitter {
                 "Only measurements of 1 and 2 dimensions are implemented yet.");
           }
 
-          if (boundParams.covariance().has_value()) {
-            trackStateProxy.predictedCovariance() =
-                std::move(*boundParams.covariance());
-          }
-
-          trackStateProxy.jacobian() = std::move(jacobian);
-          trackStateProxy.pathLength() = std::move(pathLength);
+          // Set the measurement type flag
+          typeFlags.set(TrackStateFlag::MeasurementFlag);
+          // We count the processed state
+          ++result.processedStates;
+          ACTS_VERBOSE("Actor - indices after processing, before over writing:"
+                       << "\n\t"
+                       << "result.lastMeasurementIndex: "
+                       << result.lastMeasurementIndex << "\n\t"
+                       << "trackStateProxy.index(): " << trackStateProxy.index()
+                       << "\n\t"
+                       << "result.lastTrackIndex: " << result.lastTrackIndex
+                       << "\n\t"
+                       << "currentTrackIndex: " << currentTrackIndex)
+          result.lastMeasurementIndex = currentTrackIndex;
+          result.lastTrackIndex = currentTrackIndex;
+        } else {
+          ACTS_INFO("Actor: This case is not implemented yet")
         }
       }
 
-      if (result.surfaceCount > 17) {
+      if (result.surfaceCount > 900) {
```

Comment:
Arbitrary limit? Is there no "good" abort condition?

Reply:
yes, I haven't found a better condition yet. In the next PR I will add the python framework which hopefully will help me.

Reply:
Interesting, did you actually run into this problem already?

Reply:
In the beginning I had to set it to a very small number, otherwise it would crash after the last surface, since we did handle them properly. Now it keeps recounting the last surface until it reaches the limit, since we have no other abort-condition yet.